src/goto-instrument/contracts/contracts.cpp

```diff
@@ -942,8 +942,8 @@ void code_contractst::check_frame_conditions(
       instruction_it->is_other() &&
       instruction_it->get_other().get_statement() == ID_havoc_object)
     {
-      const exprt &havoc_argument = dereference_exprt(
-        to_typecast_expr(instruction_it->get_other().operands().front()).op());
+      const auto havoc_argument =
+        pointer_object(instruction_it->get_other().operands().front());
```

Comment:
Could the commit message please be a bit more verbose? You might know what the bug exactly is/was, but someone browsing the git history at a later time most likely won't.

Reply:
Done; thanks!